test/Microsoft.ComponentDetection.Detectors.Tests/YarnLockDetectorTests.cs

```diff
@@ -240,6 +240,50 @@ public async Task WellFormedYarnLockV1WithWorkspace_FindsComponentAsync()
             parentComponent => parentComponent.Name == componentA.Name && parentComponent.Version == version0);
     }
 
+    [TestMethod]
+    public async Task WellFormedYarnLockV1WithWorkspace_CheckFilePathsAsync()
+    {
+        var directory = new DirectoryInfo(Path.GetTempPath());
+
+        var version0 = NewRandomVersion();
+        var componentA = new YarnTestComponentDefinition
+        {
+            ActualVersion = version0,
+            RequestedVersion = $"^{version0}",
+            ResolvedVersion = "https://resolved0/a/resolved",
+            Name = Guid.NewGuid().ToString("N"),
+        };
+
+        var componentStream = YarnTestUtilities.GetMockedYarnLockStream("yarn.lock", this.CreateYarnLockV1FileContent(new List<YarnTestComponentDefinition> { componentA }));
+
+        var workspaceJson = new
+        {
+            name = "testworkspace",
+            version = "1.0.0",
+            @private = true,
+            workspaces = new[] { "workspace" },
+        };
+        var str = JsonConvert.SerializeObject(workspaceJson);
+        var workspaceJsonComponentStream = new ComponentStream { Location = directory.ToString(), Pattern = "package.json", Stream = str.ToStream() };
+
+        var packageStream = NpmTestUtilities.GetPackageJsonOneRootComponentStream(componentA.Name, componentA.RequestedVersion);
+
+        var (scanResult, componentRecorder) = await this.DetectorTestUtility
+            .WithFile("yarn.lock", componentStream.Stream)
+            .WithFile("package.json", workspaceJsonComponentStream.Stream, new[] { "package.json" }, Path.Combine(Path.GetTempPath(), "package.json"))
+            .WithFile("package.json", packageStream.Stream, new[] { "package.json" }, Path.Combine(Path.GetTempPath(), "workspace", "package.json"))
+            .ExecuteDetectorAsync();
+
+        Assert.AreEqual(ProcessingResultCode.Success, scanResult.ResultCode);
+
+        var detectedComponents = componentRecorder.GetDetectedComponents();
+
+        // checking workspace's package.json entry gets added as filepath.
+        var filePathsCount = detectedComponents.First().FilePaths.Count;
+        Assert.AreEqual(1, filePathsCount);
+        Assert.AreEqual(Path.Combine(Path.GetTempPath(), "workspace", "package.json"), detectedComponents.First().FilePaths.First());
```

Comment:
I don't think we should have confidence that the order of file locations is deterministic. 

Also, we should use `FluentAssertions` for better readability.